benchmarks/benchmarks/ci/esmf_regridder.py

```diff
@@ -363,3 +330,91 @@ def time_lazy_regridding(self, cache):
     def time_regridding_realisation(self, cache):
         assert self.result.has_lazy_data()
         _ = self.result.data
+
+
+class TimeRegridderIO(MultiGridCompare):
+    def setup_cache(self):
+        from esmf_regrid.experimental.io import save_regridder
+
+        SYNTH_DATA_DIR = Path().cwd() / "tmp_data"
+        SYNTH_DATA_DIR.mkdir(exist_ok=True)
+
+        for tp in self.params:
+            (
+                lon_bounds,
+                lat_bounds,
+                n_lons_src,
+                n_lats_src,
+                n_lons_tgt,
+                n_lats_tgt,
+                _,
+                alt_coord_system,
+            ) = self.get_args(tp)
+            src_grid = _grid_cube(
+                n_lons_src,
+                n_lats_src,
+                lon_bounds,
+                lat_bounds,
+                alt_coord_system=alt_coord_system,
+            )
+            tgt_grid = _grid_cube(
+                n_lons_tgt,
+                n_lats_tgt,
+                lon_bounds,
+                lat_bounds,
+                alt_coord_system=alt_coord_system,
+            )
+            src_mesh_cube = _gridlike_mesh_cube(
+                n_lons_src,
+                n_lats_src,
+            )
+            tgt_mesh_cube = _gridlike_mesh_cube(
+                n_lons_tgt,
+                n_lats_tgt,
+            )
+
+            mesh_to_grid_regridder = MeshToGridESMFRegridder(src_mesh_cube, tgt_grid)
+            grid_to_mesh_regridder = GridToMeshESMFRegridder(src_grid, tgt_mesh_cube)
+
+            source_file_m2g = str(SYNTH_DATA_DIR.joinpath(f"m2g_source_{tp}.nc"))
+            source_file_g2m = str(SYNTH_DATA_DIR.joinpath(f"g2m_source_{tp}.nc"))
+
+            save_regridder(mesh_to_grid_regridder, source_file_m2g)
+            save_regridder(grid_to_mesh_regridder, source_file_g2m)
+        return SYNTH_DATA_DIR
+
+    def setup(self, SYNTH_DATA_DIR, tp):
+        from esmf_regrid.experimental.io import load_regridder, save_regridder
+
+        self.load_regridder = load_regridder
+        self.save_regridder = save_regridder
+
+        self.source_file_m2g = str(SYNTH_DATA_DIR.joinpath(f"m2g_source_{tp}.nc"))
+        self.source_file_g2m = str(SYNTH_DATA_DIR.joinpath(f"g2m_source_{tp}.nc"))
+        self.destination_file_m2g = str(
+            SYNTH_DATA_DIR.joinpath(f"m2g_destination_{tp}.nc")
+        )
+        self.destination_file_g2m = str(
+            SYNTH_DATA_DIR.joinpath(f"g2m_destination_{tp}.nc")
+        )
+
+        self.mesh_to_grid_regridder = load_regridder(self.source_file_m2g)
+        self.grid_to_mesh_regridder = load_regridder(self.source_file_g2m)
+
+    def teardown(self, _, tp):
+        if os.path.exists(self.destination_file_m2g):
+            os.remove(self.destination_file_m2g)
+        if os.path.exists(self.destination_file_g2m):
+            os.remove(self.destination_file_g2m)
```

Comment:
This is removing the destination file which is created during the `time_save` benchmark. The purpose of this teardown is to ensure the file system is in an identical state whenever the benchmark is run, i.e. there is no existing file being overwritten. Also, since these files are not created during `setup_cache`, i'm not sure if they would be removed automatically.

Reply:
>The purpose of this teardown is to ensure the file system is in an identical state whenever the benchmark is run

I had forgotten that `teardown()` runs between every `setup()` run, so I see now that it is indeed necessary 👍